agent/engine/docker_image_manager.go

```diff
@@ -154,6 +154,9 @@ func (imageManager *dockerImageManager) addContainerReferenceToNewImageState(con
 			Image:      sourceImage,
 			PulledAt:   time.Now(),
 			LastUsedAt: time.Now(),
+			// The PullSucceeded filed is false by default,
+			// one has to explicitly set it to be true when the pull image succeeds.
+			PullSucceeded: false,
```

Comment:
This is because even if the image pull fails, the container reference will still be updated, so we set it to be false here. We check if there is no error with image pull, we then set it to be true.